rdtools/soiling.py

```diff
@@ -820,12 +822,20 @@ def annual_soiling_ratios(stochastic_soiling_profiles, confidence_level=68.2):
         +------------------------+-------------------------------------------+
     '''
 
-    all_profiles = pd.concat(stochastic_soiling_profiles)
-    annual_groups = all_profiles.groupby(all_profiles.index.year)
+    # Create a df with each realization as a column
+    all_profiles = pd.concat(stochastic_soiling_profiles, axis=1)
+    # Weight each day by insolation
+    all_profiles_weighted = all_profiles.multiply(insolation_daily, axis=0)
+
+    # Compute the insolation-weighted soiling ration (IWSR) for each realization
+    annual_insolation = insolation_daily.groupby(insolation_daily.index.year).sum()
```

Comment:
I think there might be a bug here.  The date range of `all_profiles` doesn't necessarily match up with the date range of `insolation_daily`, which means `all_profiles_weighted` might have a bunch of nans that don't get handled appropriately in `annual_insolation`.  For example in the PVDAQ 4 notebook, `insolation_daily` starts at `2010-02-25` but the first date in `all_profiles` is `2010-03-12`.  The insolation from 2-25 to 3-12 gets counted in the annual insolation, but because there is no soiling profile data in that range, the annual sum of `all_profiles_weighted` misses the points (equivalenty, treating the range as if the system was 100% soiled) in that period.  A similar effect happens at the end of the date range because `insolation_daily` extends farther than `all_profiles`. I think this might explain why the first and last years in the notebook's annual soiling ratio results are so much lower than the other years.

So maybe reindex or mask `insolation_daily` to match `all_profiles`?  And on a related note, is there any way that nans could be present in `all_profiles`?  That would cause a similar mismatch.

Reply:
Good point. I think I'll add an `all_profiles.dropna()` to your suggestion to get rid of any nan rows. I don't think there should be any, but if there are, they should appear in every realization.

Reply:
I also added a warning and similar logic to the original Monte Carlo. Let me know what you think.